fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckService.kt

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.shopping.middleware
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import mozilla.components.browser.state.selector.selectedTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.shopping.ProductAnalysis
+import mozilla.components.support.base.log.logger.Logger
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+
+/**
+ * Service that handles the network requests for the review quality check feature.
+ */
+interface ReviewQualityCheckService {
+
+    /**
+     * Fetches the product review for the current tab.
+     *
+     * @return [ProductAnalysis] if the request succeeds, null otherwise.
+     */
+    suspend fun fetchProductReview(): ProductAnalysis?
+}
+
+/**
+ * Service that handles the network requests for the review quality check feature.
+ *
+ * @property browserStore Reference to the application's [BrowserStore].
+ */
+class ReviewQualityCheckServiceImpl(
+    private val browserStore: BrowserStore,
+) : ReviewQualityCheckService {
+
+    private val logger = Logger("ReviewQualityCheckService")
+
+    /**
+     * Fetches the product review for the current tab.
+     *
+     * @return [ProductAnalysis] if the request succeeds, null otherwise.
+     */
+    override suspend fun fetchProductReview(): ProductAnalysis? = withContext(Dispatchers.Main) {
```

Comment:
This is deliberately on the main thread so it works right now, hence the draft. When [this](https://bugzilla.mozilla.org/show_bug.cgi?id=1848008) patch lands, we can update this to the default dispatcher.

Reply:
My understanding about this was incorrect above ☝️ It still throws `IllegalThreadStateException` when called on any other dispatcher.

Does [GeckoResult.then](https://searchfox.org/mozilla-central/source/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/GeckoResult.java#567) expect to be called on the main thread so `IllegalThreadStateException` is not called? @calumozilla @owlishDeveloper 

Seems we have to `requestAnalysis` on `Dispatchers.Main` as done in [GeckoSitePermissionsStorage](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/permission/GeckoSitePermissionsStorage.kt#134) and [GeckoCookieBannersStorage](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/cookiebanners/GeckoCookieBannersStorage.kt#100)

Reply:
Yep, if we run this branch and change to Dispatchers.Default, the stack trace points to `GeckoResult.then`